services/concept_insights/ci.js

```diff
@@ -0,0 +1,422 @@
+              return node.error(message, msg);
+            }
+
+            var params = {
+              id: '/corpora/'+account_id+'/'+corpus_name+'/documents/'+document_name
+            }
+
+            switch (config.docapicall) {
+              case 'annotations':
+                params.id = params.id+'/annotations';
+                concept_insights.corpora.getDocumentAnnotations(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relatedconcepts':
+                concept_insights.corpora.getRelatedConcepts(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relationscores':
+              params.concepts = msg.concepts;
+                concept_insights.corpora.getRelationScores(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+            }
+          } else {
+            //Corpus mode
+            var params = {
+              corpus: '/corpora/'+account_id+'/'+corpus_name
+            }
+            switch (config.corpusapicall) {
+              case 'statistics':
+                concept_insights.corpora.getCorpusStats(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relatedconcepts':
+                concept_insights.corpora.getRelatedConcepts(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relationscores':
+                params.concepts = msg.concepts;
+                concept_insights.corpora.getRelationScores(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'conceptualsearch':
+                if (msg.concepts) {
+                  params.ids = msg.concepts;
+                  concept_insights.corpora.getRelatedDocuments(params, function(err,res) {
+                    handleWatsonCallback(node,msg,err,res);
+                  });
+                } else {
+                  node.status({});
+                  var message = 'No concepts exist on the msg object';
+                  node.error(message, msg);
+                }
+                break;
+
+              case 'documentlabel':
+
+                if (params.query) {
+                  params.prefix=true;
+                  params.query=config.query;
+                  concept_insights.corpora.searchByLabel(params, function(err,res) {
+                      handleWatsonCallback(node,msg,err,res);
+                  });
+                } else {
+                  node.status({});
+                  var message = 'No query term specified';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 131.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2392377031/issues/source?bid=3225115&fileBranchId=3228814#l146)